docs/reproducing-results.md

```diff
@@ -362,9 +362,59 @@ run inference through Nvidia NIM API.
        subprocess.run(cmd, shell=True, check=True)
    ```
 
-6. Now all the data is generated and you can follow up by converting it to the SFT format.
-
-   TBD
+6. Check for test set contamination.
+   We test against GSM8K, MATH, AMC 2023, and AIME 2024.  
+
+   First we use this command to get all the unique problems:
+   ```
+   cat /workspace/new-problems-solution-augmentation/*/*/output-rs0.jsonl | jq '.problem' | sort -u | shuf | jq -c -R '{"problem": .}' > /workspace/new-problems-solution-augmentation/unique_problems_all.jsonl
+   ``` 
+ 
+   Retrieve top-5 similar items from the test sets
+   ```
+   python -m nemo_skills.inference.retrieve_similar \
+      ++retrieve_from=/workspace/new-problems-solution-augmentation/unique_problems_all.jsonl \
+      ++compare_to="./nemo_skills/dataset/gsm8k/test.jsonl ./nemo_skills/dataset/math/test.jsonl ./nemo_skills/dataset/amc23/test.jsonl ./nemo_skills/dataset/aime24/test.jsonl" \
+      ++output_file=/workspace/new-problems-solution-augmentation/contamination-retrieved.jsonl \
+      ++top_k=5
+   ```
+   > **_NOTE:_** Currently the above command doesn't run inside docker, so you will need to install additional packages.
+
+   Next, you need to run LLM inference to check those closest found problems from the output file. Here is an example using Llama-405B from Nvidia API catalog, but you can replace it with OpenAI models or self-hosted models.
```

Comment:
Since this is about reproducing our results, I think we don't need to mention that so that it's clear that we used 405b. But we should put this inside decontamination doc

Reply:
Good point. Let me remove it.